bolt-sidecar/src/state/execution.rs

```diff
@@ -200,12 +207,23 @@ impl<C: StateFetcher> ExecutionState<C> {
             return Err(ValidationError::ChainIdMismatch);
         }
 
-        // Check if there is room for more commitments
+        let max_commitments_per_slot = self.max_commitments_per_slot.get();
+        let max_committed_gas_per_slot = self.max_committed_gas_per_slot.get();
+
+        // Check if there is room for more commitments and gas in the block template
         if let Some(template) = self.get_block_template(target_slot) {
-            if template.transactions_len() >= self.max_commitments_per_slot.get() {
+            if template.transactions_len() >= max_commitments_per_slot {
                 return Err(ValidationError::MaxCommitmentsReachedForSlot(
                     self.slot,
-                    self.max_commitments_per_slot.get(),
+                    max_commitments_per_slot,
+                ));
+            }
+
+            // Check if the committed gas exceeds the maximum
+            if template.committed_gas().to::<u64>() >= max_committed_gas_per_slot {
+                return Err(ValidationError::MaxCommittedGasReachedForSlot(
+                    self.slot,
+                    max_committed_gas_per_slot,
                 ));
             }
         }
```

Comment:
Here I think you should check if 
`template.committed_gas() + tx.gas_limit >= max_committed_gas_per_slot`
otherwise with the previous request I can ask for huge gas limit and overcome the max value.
Does it make sense?

Reply:
Yes good point! Then I think we should also do the `tx.gas_limit >= max_committed_gas_per_slot` check in case no template is there right? updated